src/main/java/ladysnake/effective/client/world/WaterfallCloudGenerators.java

```diff
@@ -64,13 +64,14 @@ public static void tick() {
     }
 
     private static void tickParticles(World world) {
-        for (BlockPos pos : particlesToSpawn.keySet()) {
-            if (pos != null) {
-                if (particlesToSpawn.put(pos, particlesToSpawn.getInt(pos) - 1) <= 0) {
-                    particlesToSpawn.removeInt(pos);
+        synchronized (particlesToSpawn) {
```

Comment:
I don't think this `synchronized` block does anything because the particles are not being ticked from multiple threads simultaneously.